pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -191,8 +192,13 @@ private DataTable processQueryInternal(ServerQueryRequest queryRequest, Executor
     }
 
     List<String> segmentsToQuery = queryRequest.getSegmentsToQuery();
-    List<String> missingSegments = new ArrayList<>();
-    List<SegmentDataManager> segmentDataManagers = tableDataManager.acquireSegments(segmentsToQuery, missingSegments);
+    List<String> unAcquiredSegments = new ArrayList<>();
+    List<SegmentDataManager> segmentDataManagers = tableDataManager.acquireSegments(
+        segmentsToQuery, unAcquiredSegments);
+    List<String> missingSegments =
+        unAcquiredSegments.stream()
+            .filter(segmentName -> !tableDataManager.isSegmentDeletedRecently(segmentName))
```

Comment:
Suggest moving this logic inside the if block of Line 301 so that pinot-server doesn't have to do it for all the queries all the time.